sdk/core/Azure.Core/src/DynamicData/MutableJsonChange.cs

```diff
@@ -47,18 +31,63 @@ internal struct MutableJsonChange
 
         public MutableJsonChangeKind ChangeKind { get; }
 
-        public JsonValueKind ValueKind => GetSerializedValue().ValueKind;
+        public readonly JsonValueKind ValueKind => Value switch
```

Comment:
FYI - @m-redding, this is going to affect the annotations you are adding in https://github.com/Azure/azure-sdk-for-net/pull/37734.